pahfit/model.py

```diff
@@ -284,9 +284,19 @@ def amp_guess(row, fwhm):
         else:
             loop_over_non_fixed("line", "power", lambda row: some_flux)
 
-        # set the fwhms in the features table
+        # Set the fwhms in the features table. Slightly different logic,
+        # as the fwhm for lines are masked by default
         if calc_line_fwhm:
-            loop_over_non_fixed("line", "fwhm", line_fwhm_guess, force=True)
+            for row_index in np.where(self.features["kind"] == "line")[0]:
+                row = self.features[row_index]
+                if np.ma.is_masked(row["fwhm"]):
```

Comment:
Does that work with a structured array?

Reply:
As far as I know, the element (features[row_index][col_name]) can still be masked in its entirety. I.e., it can be equal to np.ma.masked, or it can be a structured array element (the one we defined with 'val', 'min', and 'max')

Reply:
How would a structure array have just a single value, though?  I thought our method for "fully masking" a given parameter is to mask the value?

Reply:
E.g. 
```
In [12]: f.loc['PAH_6.2']['wavelength']
Out[12]: (6.22, nan, nan)

# "starlight" does not have the "wavelength" parameter -> masked
In [9]: f.loc['starlight']['wavelength']
Out[9]: masked

In [14]: np.ma.is_masked(f.loc['starlight']['wavelength'])
Out[14]: True
```

Reply:
I think that's just for display.  Underlying that is still a structure array member.  I believe masking _any_ of the structure array members results in this.

Reply:
I ran into a bug due to this. Apparently, in our current setup, `row['fwhm']` can be either 0-dimensional or 1-dimensional. (with `row` a row of the features table)

This is a problem, because if we want to check if 'fwhm' is masked, then `row['fwhm'][0]` will yield an error if `row['fwhm']` is 0-dimensional. Vice versa, in the 1-dimensional case, we also get an error if we try to use `np.ma.is_masked(row['fwhm'])`.

A workaround would be checking `row['fwhm'].ndim` each time.

Reply:
To be clear, if the above displays `masked` then we have the 0-dimensional case. So yes, it really is a single value somehow. Even more striking: if I check the data attribute, i get this
```
ipdb>  p row['fwhm'].data

array(0.)
```